R/zchunk_L225.hydrogen.R

```diff
@@ -1,237 +1,287 @@
+    int_years <- c(BASE_YEARS, FUTURE_YEARS)[which(!(c(BASE_YEARS, FUTURE_YEARS) %in% df_years))]
+
+    A25.globaltech_shrwt_raw_long %>%
+      select(-share.weight,-year) %>%
+      unique() %>%
+      repeat_add_columns(tibble::tibble(year=int_years)) %>%
+      bind_rows(A25.globaltech_shrwt_raw_long) %>%
+      group_by(supplysector, subsector, technology) %>%
+      mutate(share.weight=approx_fun(year, share.weight, rule=1)) %>%
+      filter(year %in% c(BASE_YEARS,FUTURE_YEARS)) %>%
+      rename(sector.name=supplysector) %>% #Assign the columns "sector.name" and "subsector.name", consistent with the location info of a global technology
+      rename(subsector.name=subsector) ->
+      L225.GlobalTechShrwt_h2
+
+    # L225.PrimaryRenewKeyword_h2: Keywords of primary renewable electric generation technologies
+    A25.globaltech_keyword %>%
+      repeat_add_columns(tibble::tibble(year=c(BASE_YEARS, FUTURE_YEARS))) %>%
+      rename(sector.name=supplysector) %>%
+      rename(subsector.name=subsector) %>%
+      filter(!is.na(primary.renewable)) %>%
+      select(LEVEL2_DATA_NAMES[['GlobalTechYr']], 'primary.renewable') ->
+      L225.PrimaryRenewKeyword_h2
+
+    # L225.GlobalTechCapture_h2: CO2 capture fractions from global fertilizer production technologies with CCS
+    # Note: No need to consider historical periods or intermittent technologies here
+    A25.globaltech_co2capture %>%
+      gather(year, remove.fraction, -supplysector, -subsector, -technology) %>%
+      mutate(year=as.numeric(year)) ->
+      A25.globaltech_co2capture_raw_long
+
+    df_years <- unique(A25.globaltech_co2capture_raw_long$year)
+    int_years <- FUTURE_YEARS[which(!(FUTURE_YEARS %in% df_years))]
+
+    A25.globaltech_co2capture_raw_long %>%
+      select(-remove.fraction,-year) %>%
+      unique() %>%
+      repeat_add_columns(tibble::tibble(year=int_years)) %>%
+      bind_rows(A25.globaltech_co2capture_raw_long) %>%
+      group_by(supplysector, subsector, technology) %>%
+      mutate(remove.fraction=approx_fun(year, remove.fraction, rule=1)) %>%
+      mutate(remove.fraction=round(remove.fraction,energy.DIGITS_REMOVE.FRACTION)) %>%
+      filter(year %in% FUTURE_YEARS) %>%
+      rename(sector.name=supplysector) %>% #Assign the columns "sector.name" and "subsector.name", consistent with the location info of a global technology
+      rename(subsector.name=subsector) %>%
+      select(LEVEL2_DATA_NAMES[['GlobalTechYr']], 'remove.fraction') %>%
+      mutate(storage.market=energy.CO2.STORAGE.MARKET) ->
+      L225.GlobalTechCapture_h2
+
+    # ===================================================
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
+
+    tibble() %>%
```

Comment:
Here and below--this is passing an empty data frame in, which is **not** what you want! Replace with the names of the final output data frames. If this is confusing, let's talk on Monday--I'm happy to clarify.